core/tauri/src/window.rs

```diff
@@ -1330,6 +1352,17 @@ impl<R: Runtime> Window<R> {
             manager.package_info(),
           );
         } else if payload.cmd.starts_with("plugin:") {
+          if !is_local {
+            let command = invoke.message.command.replace("plugin:", "");
+            let plugin_name = command.split('|').next().unwrap().to_string();
+            if !scope
+              .map(|s| s.plugins.contains(&plugin_name))
+              .unwrap_or(true)
+            {
+              invoke.resolver.reject("Plugin not allowed");
+              return Ok(());
+            }
+          }
           manager.extend_api(invoke);
         } else {
           manager.run_invoke_handler(invoke);
```

Comment:
Since we forked Tauri to ship our patched version, I had some ideas. I think whitelisting  which command can be run would be really cool, but not a blocker for this PR.

Reply:
See the message above. Also, it would need a lot of code to add this :(

Reply:
This can come in a subsequent PR, happy to pitch in for the implementation